packages/core/src/browser/label-provider.ts

```diff
@@ -88,6 +89,41 @@ export interface DidChangeLabelEvent {
     affects(element: object): boolean;
 }
 
+// copied and modified from https://github.com/microsoft/vscode/blob/1.44.2/src/vs/platform/label/common/label.ts#L31-L33
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+export interface FormatterChangeEvent {
+    scheme: string;
+}
+
+// copied from https://github.com/microsoft/vscode/blob/1.44.2/src/vs/platform/label/common/label.ts#L35-L40
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+export interface ResourceLabelFormatter {
+    scheme: string;
+    authority?: string;
+    priority?: boolean;
+    formatting: ResourceLabelFormatting;
+}
+
+// copied from https://github.com/microsoft/vscode/blob/1.44.2/src/vs/platform/label/common/label.ts#L42-L49
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+export interface ResourceLabelFormatting {
+    label: string; // myLabel:/${path}
+    separator: '/' | '\\' | '';
+    tildify?: boolean;
+    normalizeDriveLetter?: boolean;
+    workspaceSuffix?: string;
```

Comment:
it is never used, did we forget some logic?

Reply:
Removed the leftovers. In vscode it is used for the [`getWorkspaceLabel()`](https://github.com/microsoft/vscode/blob/1.44.2/src/vs/workbench/services/label/common/labelService.ts#L178) function, but we are using only the `getLongName()` function

Reply:
I think `WorkspaceUriLabelProviderContribution.getLongName` should respect it. That's a reason why we introduced workspace aware provider.